File: week09/week09-2.c
int diagonalSum(int** a, int aSize, int* aColSize) {
    int n = aSize;
    int ans = 0;
    for (int i = 0; i < n; i++) {
        ans += a[i][i];
        ans += a[i][n - 1 - i];
    }
    if (n % 2)
        ans -= a[n / 2][n / 2];
    return ans;
}

File: week09/week09-2.py
class Solution:
    def diagonalSum(self, a: List[List[int]]) -> int:
        n=len(a)
        ans=0
        for i in range(n):
            ans+=a[i][i]
            ans+=a[i][-i-1]
        if n%2:ans-=a[n//2][n//2]
        return ans